src/main/java/hudson/util/VersionNumber.java

```diff
@@ -525,6 +528,32 @@ public int digit(int idx) {
         return ((IntegerItem) item).value.intValue();
     }
 
+    /**
+     * Returns a digit (numeric component) by its position. Once a non-numeric component is found all remaining components
+     * are also considered non-numeric by this method.
+     *
+     * @param idx Digit position we want to retrieve starting by 0.
+     * @return The digit or -1 in case the position does not correspond with a digit.
+     */
+    public int getDigitAt(int idx) {
+        if (idx < 0) {
```

Comment:
it would seem more appropriate to throw an `IllegalArgumentException` for this case.